src/main/webapp/user_profile_script.js

```diff
@@ -12,66 +12,77 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-async function getInfo(keyString) {
+
+
+async function getTaskInfo(keyString) {
     const queryURL = "/tasks/info?key=" + keyString;
     const request = new Request(queryURL, {method: "GET"});
     const response = await fetch(request);
     const info = await response.json();
-    console.log(info);
     return info;
 }
 
 async function deleteTask(keyString) {
-    const info = await getInfo(keyString);
+    const info = await getTaskInfo(keyString);
     if (info.status !== "OPEN") {
         window.alert("You can only delete an 'OPEN' task.")
     } else {
         if (confirm("Are you sure that you want to delete the task?")) {
             const queryURL = "/tasks?key=" + keyString;
             const request = new Request(queryURL, {method: "DELETE"});
             const response = await fetch(request);
+            showNeedHelp();
         }
     }
 }
 
 async function editTask(keyString) {
-    const info = await getInfo(keyString);
-    console.log(info);
+    const info = await getTaskInfo(keyString);
     if (info.status !== "OPEN") {
         window.alert("You can only edit an 'OPEN' task.")
     } else {
-        document.getElementById("edit-content-input").value = info.detail;
+        document.getElementById("edit-detail-input").value = info.detail;
         document.getElementById("edit-point-input").value = info.reward.toString();
         const id_input = document.getElementById("task-id-input");
         id_input.value = info.keyString;
-        id_input.readOnly = true;
-        id_input.style.display = "none"
-        document.getElementById("editTaskModal").style.display = "block";
+        document.getElementById("editTaskModalWrapper").style.display = "block";
+        showNeedHelp()
     }
 }
 
 async function completeTask(keyString) {
-    const info = await getInfo(keyString);
+    const info = await getTaskInfo(keyString);
     if (info.status !== "IN PROGRESS") {
         window.alert("You have already marked the task as complete.");
+    } else if (info.owner === info.helper) {
+        window.alert("You cannot complete a task published by yourself! The task will now be removed from the system!");
+        var queryURL = "/tasks/info?key=" + keyString + "&status=" + "OPEN";
+        var request = new Request(queryURL, {method: "POST"});
+        var response = await fetch(request);
+        queryURL = "/tasks?key=" + keyString;
+        request = new Request(queryURL, {method: "DELETE"});
+        response = await fetch(request);
```

Comment:
It seems rather harsh to delete the task if the user manages to try marking it complete. Shouldn't we just reject the attempt, display the first half of the alert, and call it good?

Reply:
Ah yeah. I could do that, making the user have no choice but to abandon the task at the end. I will fix this in the new commit.